environment-dev.yml

```diff
@@ -39,7 +39,7 @@ dependencies:
   - pymssql
   - psycopg2
   - pip:
-      - xgboost>=1.3
+      - xgboost>=1.5.0
```

Comment:
This is a relatively recent version of XGBoost. Don't we want to allow the user to use older versions of XGBoost. Maybe we should get rid of behavior specific for versions >=1.5.0?

Reply:
We yet stick to the strategy that Modin is aligned with the latest version of the packages it mimics (i.e. pandas), the same should be true and for XGBoost.

Maybe we could do version-specific behaviour for XGBoost after #4301 is merged, but for now, I'm voting up for updating the pin.

Reply:
I agree with @dchigarev. We can start support of multiple xgb versions after strategy in #4301 will be applied.

> This is a relatively recent version of XGBoost. Don't we want to allow the user to use older versions of XGBoost. Maybe we should get rid of behavior specific for versions >=1.5.0?

We can't get rid from specific behavior for versions >=1.5.0 because starting from 1.5 xgboost significantly changed exception types. I propose to update pin for now.

Reply:
That makes sense to me! Let's update the pin. @prutskov, please move the release note to `release_notes-0.15.0.rst`.

Reply:
@YarShev, done.